### ISSUE TITLE
issues running rqcfilter.wdl v1.0.14 with JAWS

### ISSUE THREAD

**Issue by aclum** [microbiomedata/nmdc_automation#641]
related to #639 

This wdl is valid with jaws if you submit on the command line with the wdl directly & the files are present locally, last tested today with jaws job id 147028. This wdl does not work with automation + the jaws API which passes a bundle.zip with the subworkflows.

Below is the error from the JAWS API, it appears to be an issue with the required but empty arrays.

```
2025-11-06 13:44:49,024 INFO: Claiming job nmdc:sys02snn8989
2025-11-06 13:44:49,240 INFO: Prepare and cache new job: nmdc:sys0wg5tgf81
2025-11-06 13:44:50,089 INFO: WDL file: /tmp/tmpst7p915b.wdl
2025-11-06 13:44:50,089 INFO: Bundle file: /tmp/tmpt1oprb03.zip
2025-11-06 13:44:50,089 INFO: Workflow inputs:
2025-11-06 13:44:50,089 INFO: {
  "rqcfilter.input_files": [
    "https://data.microbiomedata.org/data/nmdc:dgns-11-zayj8368/nmdc:dgns-11-zayj8368/11862.3.224569.CGCTTAA-GTTAAGC.fastq.gz"
  ],
  "rqcfilter.proj": "nmdc:wfrqc-15-synp3094.1",
  "rqcfilter.input_fq1": "[]",
  "rqcfilter.input_fq2": "[]",
  "rqcfilter.shortRead": true,
  "rqcfilter.interleaved": true
}
2025-11-06 13:44:50,089 INFO: Workflow labels:
2025-11-06 13:44:50,089 INFO: {
  "release": "v1.0.14",
  "wdl": "rqcfilter.wdl",
  "git_repo": "https://github.com/microbiomedata/ReadsQC",
  "submitter": "nmdcda",
  "pipeline_version": "v1.0.14",
  "pipeline": "rqcfilter.wdl",
  "activity_id": "nmdc:wfrqc-15-synp3094.1",
  "opid": "nmdc:sys0wg5tgf81"
}
2025-11-06 13:44:55,817 ERROR: Failed to Submit Job: {'message': 'Subworkflows not found. Error: Failed to evaluate input \'input_fq1\' (reason 1
 of 1): No coercion defined from \'"[]"\' of type \'spray.json.JsString\' to \'Array[String]\'.\nFailed to evaluate input \'input_fq2\' (reason 1
 of 1): No coercion defined from \'"[]"\' of type \'spray.json.JsString\' to \'Array[String]\'.\n'}
2025-11-06 13:44:55,818 ERROR: Error removing directory /tmp: [Errno 13] Permission denied: '/tmp'
2025-11-06 13:45:00,358 INFO: WDL file: /tmp/tmp9qp_c22d.wdl
2025-11-06 13:45:00,358 INFO: Bundle file: /tmp/tmp8y_nbype.zip
2025-11-06 13:45:00,358 INFO: Workflow inputs:
2025-11-06 13:45:00,358 INFO: {
  "rqcfilter.input_files": [
    "https://data.microbiomedata.org/data/nmdc:dgns-11-zayj8368/nmdc:dgns-11-zayj8368/11862.3.224569.CGCTTAA-GTTAAGC.fastq.gz"
  ],
  "rqcfilter.proj": "nmdc:wfrqc-15-synp3094.1",
  "rqcfilter.input_fq1": "[]",
  "rqcfilter.input_fq2": "[]",
  "rqcfilter.shortRead": true,
  "rqcfilter.interleaved": true
}
2025-11-06 13:45:00,358 INFO: Workflow labels:
2025-11-06 13:45:00,358 INFO: {
  "release": "v1.0.14",
  "wdl": "rqcfilter.wdl",
  "git_repo": "https://github.com/microbiomedata/ReadsQC",
  "submitter": "nmdcda",
  "pipeline_version": "v1.0.14",
  "pipeline": "rqcfilter.wdl",
  "activity_id": "nmdc:wfrqc-15-synp3094.1",
  "opid": "nmdc:sys0wg5tgf81"
}
2025-11-06 13:45:02,776 ERROR: Failed to Submit Job: {'message': 'Subworkflows not found. Error: Failed to evaluate input \'input_fq1\' (reason 1
 of 1): No coercion defined from \'"[]"\' of type \'spray.json.JsString\' to \'Array[String]\'.\nFailed to evaluate input \'input_fq2\' (reason 1
 of 1): No coercion defined from \'"[]"\' of type \'spray.json.JsString\' to \'Array[String]\'.\n'}
2025-11-06 13:45:02,777 ERROR: Error removing directory /tmp: [Errno 13] Permission denied: '/tmp'
Traceback (most recent call last):
  File "/global/cfs/cdirs/m3408/nmdc_automation/dev/nmdc_automation/.venv/lib/python3.11/site-packages/tenacity/__init__.py", line 480, in __call
__
```

This does kill the watcher.

**Comment by aclum** [comment]
This error is from the inputs.json having double quotes around the empty array. The linked PR updates the workflows.yaml uses a string without double quotes but that causes a number of the workflow automation tests to fail. @jasmynp will look at this next week and work with @vlilanl to determine if we should change the WDL to make these variables optional in the inputs.json, which involves a number of workarounds b/c WDL doesn't directly support OR operations for inputs or if we will update nmdc_automation code to handle these empty arrays.

**Comment by jasmynp** [comment]
@vlilanl was able to update the wdl and we both had successful tests run on Perlmutter. She currently has tests running on JAWS.

**Comment by ssarrafan** [comment]
Moving to new sprint. @vlilanl isn't working today. Appears to be almost done.
